hnn_core/cell.py

```diff
@@ -341,6 +364,41 @@ def gid(self, gid):
         else:
             raise RuntimeError('Global ID for this cell already assigned!')
 
+    def distance_section(self, target_sec_name, curr_node):
+        # Python version of the Neuron distance function
+        # https://nrn.readthedocs.io/en/latest/python/modelspec/programmatic/topology/geometry.html#distance  # noqa
+        if self.cell_tree is None:
+            raise TypeError("distance_section() "
+                            "cannot work with cell_tree as None.")
+        if curr_node not in self.cell_tree:
+            return 1000000
+
+        # Children of the current section
+        curr_sec_children = self.cell_tree[curr_node]
+        # All sections have 0 and 1 ends
+        end_pts = (0, 1)
+
+        # Base condition
+        # If target section is connected to current section
+        # Return (target section length / 2)
+        # As distances are measured till the centre of the target section
+        for end_pt in end_pts:
+            if (target_sec_name, end_pt) in curr_sec_children:
+                return self.sections[target_sec_name].L / 2
+
+        dist = 1000000  # Return large value
+
+        # Recursion to find distance
+        for node in self.cell_tree[curr_node]:
+            if (node[0] == curr_node[0]):
+                dist_temp = (self.distance_section(target_sec_name, node) +
+                             self.sections[node[0]].L)
+            else:
+                dist_temp = self.distance_section(target_sec_name, node)
+            dist = min(dist, dist_temp)
```

Comment:
```suggestion
            dist = np.nanmin(dist, dist_temp)
```

Reply:
It throws a warning when all nan values are passed to the function. The results are correct. This warning is fine right?

Reply:
For now I have added a check to resolve this issue.